File: DatalakeClient/src/@types/enums/TagHistoryUse.ts
export enum TagHistoryUse {
	Initial = 0,
	Basic = 1,
	Aggregated = 2,
	Continuous = 3,
	Outdated = 100,
}

export function TagHistoryUseDescription(type: TagHistoryUse) {
	switch (type) {
		case TagHistoryUse.Initial: return 'начальное'
		case TagHistoryUse.Basic: return 'полученное'
		case TagHistoryUse.Aggregated: return 'вычисленное'
		case TagHistoryUse.Continuous: return 'протянутое'
		case TagHistoryUse.Outdated: return 'устаревшее'
		default: return '?'
	}
}

File: Server/Services/Receiver/Models/OldDatalake/TagHistoryUse.cs
namespace Datalake.Server.Services.Receiver.Models.OldDatalake;

/// <summary>
/// Флаг использования
/// </summary>
public enum TagHistoryUse
{
	/// <summary>
	/// Начальное значение
	/// </summary>
	Initial = 0,
	/// <summary>
	/// Основное значение
	/// </summary>
	Basic = 1,
	/// <summary>
	/// Результат агрегирования
	/// </summary>
	Aggregated = 2,
	/// <summary>
	/// Протянутое значение
	/// </summary>
	Continuous = 3,
	/// <summary>
	/// Устаревшее значение
	/// </summary>
	Outdated = 100,
	/// <summary>
	/// Значение не существует
	/// </summary>
	NotFound = 101,
}
